### PR TITLE
Fixes #33840 - Add pagination to all roles modal

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @xprazak2 👍 overall works well

a page transition creates a loading state for the entire page instead of just the table:
https://user-images.githubusercontent.com/11807069/140185745-2e0815b0-5f67-460c-87b4-28e8ce55baee.mov

same in modal, also a fixed hight value can prevent the modal's size change while loading:
https://user-images.githubusercontent.com/11807069/140185956-3452bfc1-c2af-453c-99fb-49b0adc35ad6.mov

File: webpack/components/AnsibleHostDetail/components/RolesTab/AllRolesModal/AllRolesTable.js [hunk 1]
```diff
@@ -11,40 +12,78 @@ import {
...
+            {columns.map(col => (
+              <Th key={`${col}-all`}>{col}</Th>
+            ))}
           </Tr>
-        ))}
-      </Tbody>
-    </TableComposable>
+        </Thead>
+        <Tbody>
+          {allAnsibleRoles.map(role => (
```
[the remark this answers is not in the document]
Comment (xprazak2) [reply webpack/components/AnsibleHostDetail/components/RolesTab/AllRolesModal/AllRolesTable.js:65]: Which values should I extract?

**Comment by xprazak2** [comment]
Thank you for review & testing @amirfefer! Any suggestions to what height the modal size should be fixed?

Scope here is just to add pagination into modal, I can adjust the loading states as a follow-up (since that will be probably same for all the tables) if that is something we want.

**Comment by amirfefer** [comment]
PF design [documentation](https://www.patternfly.org/v4/components/modal/design-guidelines) doesn't mention fixed height for modals, @MariSvirik, any idea?

File: webpack/components/AnsibleHostDetail/components/RolesTab/AllRolesModal/AllRolesTable.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amirfefer) [reply webpack/components/AnsibleHostDetail/components/RolesTab/AllRolesModal/AllRolesTable.js:65]: {id, name, inherited }

**Review by amirfefer — APPROVED** [review]
let's fix the loading and css in a follow-up PR :), the pagination itself works as expected

**Review by ezr-ondrej — APPROVED** [review]
Thanks @xprazak2 and @amirfefer ! :+1:

**Pull request merged**